pkg/cluster/loadbalancerprofile.go

```diff
@@ -0,0 +1,386 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+	"github.com/Azure/ARO-RP/pkg/util/uuid"
+)
+
+func (m *manager) reconcileLoadBalancerProfile(ctx context.Context) error {
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType != api.OutboundTypeLoadbalancer {
+		return nil
+	}
+
+	resourceGroupName := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	infraID := m.doc.OpenShiftCluster.Properties.InfraID
+
+	lb, err := m.loadBalancers.Get(ctx, resourceGroupName, infraID, "")
+	if err != nil {
+		return err
+	}
+
+	err = m.reconcileOBRuleV4OBIPs(ctx, lb)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile the outbound rule "outbound-rule-v4" frontend IP Config.
+func (m *manager) reconcileOBRuleV4OBIPs(ctx context.Context, lb mgmtnetwork.LoadBalancer) error {
+	m.log.Info("reconciling outbound-rule-v4")
+	defer func() {
+		err := m.deleteUnusedManagedIPs(ctx)
+		if err != nil {
+			m.log.Error("failed to cleanup unused managed IPs, error: %w", err)
+		}
+	}()
+
+	resourceGroupName := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	infraID := m.doc.OpenShiftCluster.Properties.InfraID
+	originalOutboundIPs := getOutboundIPsFromLB(lb)
+	// ensure effectiveOutboundIPs is patched the first time running against a cluster
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.EffectiveOutboundIPs == nil {
+		err := m.patchEffectiveOutboundIPs(ctx, originalOutboundIPs)
+		if err != nil {
+			return err
+		}
+	}
+
+	desiredOutboundIPs, err := m.getDesiredOutboundIPs(ctx)
+	if err != nil {
+		return err
+	}
+
+	if areResourceRefsEqual(desiredOutboundIPs, originalOutboundIPs) {
+		return nil
+	}
+
+	// rebuild outbound-rule-v4 frontend ip config with desired outbound ips
+	removeOutboundIPsFromLB(lb)
+	addOutboundIPsToLB(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, lb, desiredOutboundIPs)
+
+	err = m.loadBalancers.CreateOrUpdateAndWait(ctx, resourceGroupName, infraID, lb)
+	if err != nil {
+		return err
+	}
+
+	// update database with new effective outbound IPs
+	err = m.patchEffectiveOutboundIPs(ctx, desiredOutboundIPs)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Remove all frontend ip config in use by outbound-rule-v4.  Frontend IP config that is used by load balancer rules will be saved.
+func removeOutboundIPsFromLB(lb mgmtnetwork.LoadBalancer) {
+	// get all outbound rule fip config to remove
+	var obRuleV4FIPConfigMap = make(map[string]mgmtnetwork.SubResource)
```

Comment:
nit: `ConfigMap` has other meanings in our industry so this is slightly confusing, even something like `ConfigsMap` is a little clearer IMO, or you can stick with just `Configs`, since golang will give us the struct type.